File: app-android/app/src/main/java/com/health2u/domain/repository/EmergencyContactRepository.kt
package com.health2u.domain.repository

import com.health2u.data.Result
import com.health2u.domain.model.EmergencyContact
import kotlinx.coroutines.flow.Flow

interface EmergencyContactRepository {
    suspend fun getEmergencyContacts(): Result<List<EmergencyContact>>
    suspend fun getEmergencyContactById(id: String): Result<EmergencyContact>
    suspend fun addEmergencyContact(contact: EmergencyContact): Result<EmergencyContact>
    suspend fun updateEmergencyContact(contact: EmergencyContact): Result<EmergencyContact>
    suspend fun deleteEmergencyContact(id: String): Result<Unit>
    fun observeEmergencyContacts(): Flow<List<EmergencyContact>>
}

File: app-ios/Sources/Health2u/Domain/Repository/EmergencyContactRepository.swift
import Foundation

public protocol EmergencyContactRepository: Sendable {
    func getEmergencyContacts() async -> Result<[EmergencyContact], APIError>
    func getEmergencyContact(id: String) async -> Result<EmergencyContact, APIError>
    func addEmergencyContact(_ contact: EmergencyContact) async -> Result<EmergencyContact, APIError>
    func updateEmergencyContact(_ contact: EmergencyContact) async -> Result<EmergencyContact, APIError>
    func deleteEmergencyContact(id: String) async -> Result<Void, APIError>
    func observeEmergencyContacts() -> AsyncStream<[EmergencyContact]>
}
